File: LeetCode/java/1791_Find_Center_of_Star_Graph.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int findCenter(int[][] edges) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();

        for (int[] edge : edges) {
            map.put(edge[0], map.getOrDefault(edge[0], 0) + 1);
            map.put(edge[1], map.getOrDefault(edge[1], 0) + 1);
        }

        for (int key : map.keySet()) {
            if (map.get(key) == edges.length) {
                answer = key;
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/1791_Find_Center_of_Star_Graph.cpp
class Solution {
public:
	int findCenter(vector<vector<int>>& edges) {
		int answer = 0;
		unordered_map<int, int> um;

		for (vector<int> edge : edges) {
			um[edge[0]]++;
			um[edge[1]]++;
		}

		for (auto m : um) {
			if (m.second == edges.size()) {
				answer = m.first;
			}
		}

		return answer;
	}
};